estimagic/optimization/optimize.py

```diff
@@ -34,51 +45,71 @@ def maximize(
     constraints=None,
     general_options=None,
     algo_options=None,
+    gradient=None,
+    gradient_options=None,
+    logging=DEFAULT_DATABASE_NAME,
+    log_options=None,
     dashboard=False,
     db_options=None,
 ):
-    """
-    Maximize *criterion* using *algorithm* subject to *pc* and bounds.
+    """Maximize *criterion* using *algorithm* subject to *constraints* and bounds.
+
+    Each argument except for ``general_options`` can also be replaced by a list of
+    arguments in which case several optimizations are run in parallel. For this, either
+    all arguments must be lists of the same length, or some arguments can be provided
+    as single arguments in which case they are automatically broadcasted.
 
     Args:
-        criterion (function):
+        criterion (function or list of functions):
             Python function that takes a pandas DataFrame with parameters as the first
             argument and returns a scalar floating point value.
 
-        params (pd.DataFrame):
+        params (pd.DataFrame or list of pd.DataFrames):
             See :ref:`params`.
 
-        algorithm (str):
+        algorithm (str or list of strings):
             specifies the optimization algorithm. See :ref:`list_of_algorithms`.
 
-        criterion_kwargs (dict):
+        criterion_kwargs (dict or list of dicts):
             additional keyword arguments for criterion
 
-        constraints (list):
+        constraints (list or list of lists):
             list with constraint dictionaries. See for details.
 
         general_options (dict):
-            additional configurations for the optimization.
+            additional configurations for the optimization
 
-        algo_options (dict):
-            algorithm specific configurations for the optimization.
+        algo_options (dict or list of dicts):
+            algorithm specific configurations for the optimization
+
+        gradient (callable or None):
+            Gradient function.
```

Comment:
If I understand `create_internal_gradient` correctly, `gradient` has to be a function that calculates a gradient numerically, given a function. But then it is not necessary that the user can change this function because in a few weeks we will just get a really good numerical gradient function. 

Thus I would suggest that for now we remove the gradient argument from the user interface and only keep gradient options (which contains arguments that influence the calculation of the numerical gradient). Then we always do what you currently do if `gradient=None`.

In the long run we can add a gradient argument, but this would then not be a numerical derivative, but a closed form gradient. However, to make this work, we need to solve #56 which should be relatively easy since we re-wrote the reparametrizations.

Reply:
I changed it such that ``process_optimizations_arguments`` always passes ``gradient = None``. Thus, the keyword is already present on lower levels, but hidden from the interface. Is this what you suggested?

Reply:
Yes, exactly. Thanks.